### PR TITLE
randomness #9: smoke tests from randomnet

### PR DESCRIPTION
**Pull request by zjma**

### CODE REVIEW

File: aptos-move/move-examples/on_chain_dice/sources/dice.move [hunk 1]
```diff
@@ -0,0 +1,21 @@
+module module_owner::dice {
+    use std::signer::address_of;
+    use std::vector;
+    use aptos_framework::randomness;
+
+    struct DiceRollHistory has key {
+        rolls: vector<u64>,
+    }
+
+    entry fun roll(account: signer) acquires DiceRollHistory {
```
Comment (danielxiangzl) [inline aptos-move/move-examples/on_chain_dice/sources/dice.move:10]: needs private entry?

File: testsuite/smoke-test/src/randomness/validator_restart_during_dkg.rs [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    let validator_clients: Vec<Client> =
+        swarm.validators().map(|node| node.rest_client()).collect();
+    let dkg_session_1 = wait_for_dkg_finish(&validator_clients[3], None, time_limit_secs).await;
+
+    info!(
+        "Current epoch is {}.",
+        dkg_session_1.metadata.dealer_epoch + 1
+    );
+
+    info!("Inject fault to all validators so they get stuck upon the first DKG message received.");
```
Comment (danielxiangzl) [inline testsuite/smoke-test/src/randomness/validator_restart_during_dkg.rs:63]: Which epoch do the validators crash? If it is any epoch other than the first epoch, this test also tested the crash recovery of randomness manager right?

**Review by danielxiangzl — APPROVED** [review]

File: aptos-move/vm-genesis/src/lib.rs [hunk 3]
```diff
@@ -421,8 +434,9 @@ fn initialize(
     );
 }
 
-fn initialize_features(session: &mut SessionExt) {
-    let features: Vec<u64> = FeatureFlag::default_features()
+fn initialize_features(session: &mut SessionExt, features_override: Option<Vec<FeatureFlag>>) {
```
Comment (zekun000) [inline aptos-move/vm-genesis/src/lib.rs:437]: this is not individual override but the whole vec?

**Review by zekun000 — APPROVED** [review]

File: aptos-move/vm-genesis/src/lib.rs (hunk 3 above)
Comment (zjma) [reply aptos-move/vm-genesis/src/lib.rs:437]: specifying a patch seems more complicated?

File: testsuite/smoke-test/src/randomness/validator_restart_during_dkg.rs (hunk 2 above)
Comment (zjma) [reply testsuite/smoke-test/src/randomness/validator_restart_during_dkg.rs:63]: we have `wait_for_all_nodes_to_catchup_to_epoch(2, ..)` so yes, also kinda covered rand_manager crash recovery.

File: aptos-move/vm-genesis/src/lib.rs (hunk 3 above)
Comment (zekun000) [reply aptos-move/vm-genesis/src/lib.rs:437]: yeah just the name is a bit confusing

File: aptos-move/move-examples/on_chain_dice/sources/dice.move (hunk 1 above)
Comment (zjma) [reply aptos-move/move-examples/on_chain_dice/sources/dice.move:10]: i thought this is private entry already?

**Pull request merged**